task/buildah/0.1/buildah.yaml

```diff
@@ -15,350 +16,354 @@ spec:
-    name: CONTEXT
-    type: string
-  - default: "true"
-    description: Verify the TLS on the registry endpoint (for push/pull to a non-TLS registry)
-    name: TLSVERIFY
-    type: string
-  - description: unused, should be removed in next task version
-    name: DOCKER_AUTH
-    type: string
-    default: ""
-  - default: "false"
-    description: Determines if build will be executed without network access.
-    name: HERMETIC
-    type: string
-  - default: ""
-    description: In case it is not empty, the prefetched content should be made available to the build.
-    name: PREFETCH_INPUT
-    type: string
-  - default: ""
-    description: Delete image tag after specified time. Empty means to keep the image tag. Time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
-    name: IMAGE_EXPIRES_AFTER
-    type: string
-  - name: COMMIT_SHA
-    description: The image is built from this commit.
-    type: string
-    default: ""
+    - description: Reference of the image buildah will produce.
+      name: IMAGE
+      type: string
+    - default: quay.io/redhat-appstudio/buildah:v1.31.0@sha256:34f12c7b72ec2c28f1ded0c494b428df4791c909f1f174dd21b8ed6a57cf5ddb
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      # also per direction from Ralph Bean, we want to use image digest based tags to use a cue to automation like dependabot or renovatebot to periodially submit pull requests that update the digest as new images are released.
+      description: The location of the buildah builder image.
+      name: BUILDER_IMAGE
+      type: string
+    - default: ./Dockerfile
+      description: Path to the Dockerfile to build.
+      name: DOCKERFILE
+      type: string
+    - default: .
+      description: Path to the directory to use as context.
+      name: CONTEXT
+      type: string
+    - default: "true"
+      description: Verify the TLS on the registry endpoint (for push/pull to a non-TLS registry)
+      name: TLSVERIFY
+      type: string
+    - description: unused, should be removed in next task version
+      name: DOCKER_AUTH
+      type: string
+      default: ""
+    - default: "false"
+      description: Determines if build will be executed without network access.
+      name: HERMETIC
+      type: string
+    - default: ""
+      description: In case it is not empty, the prefetched content should be made available to the build.
+      name: PREFETCH_INPUT
+      type: string
+    - default: ""
+      description: Delete image tag after specified time. Empty means to keep the image tag. Time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
+      name: IMAGE_EXPIRES_AFTER
+      type: string
+    - name: COMMIT_SHA
+      description: The image is built from this commit.
+      type: string
+      default: ""
+    - name: ARTIFACTS
```

Comment:
We probably want to create a new version of each Task that now task the ARTIFACTS parameter or emits an artifact result.

Reply:
> We probably want to create a new version of each Task that now task the ARTIFACTS parameter or emits an artifact result.

I'm not sold on this, if ARTIFACTS was made optional the backward compatibility would be maintained.